### PR TITLE
Replace architecture diagram with PlantUML rendered version

### PR DESCRIPTION
**Pull request by AlCutter**

### CODE REVIEW

File: binary_transparency/firmware/docs/design/README.md [hunk 1]
```diff
@@ -129,7 +129,55 @@ The design for the demo consists of a number of different entities which play
 the roles described in the claimant model above, these are shown in the
 following diagram:
 
-![overview diagram](./overview.svg)
+![ov](./ov.svg)
+<div style="display: none">
```
Comment (mhutchinson) [inline binary_transparency/firmware/docs/design/README.md:133]: It doesn't appear that this is being respected by the diff tool or previews, i.e. the following shows this startuml text in the doc.

https://github.com/google/trillian-examples/blob/4d4f715614e35697de076df8c9b775a5507c3a0e/binary_transparency/firmware/docs/design/README.md

File: binary_transparency/firmware/docs/design/README.md [hunk 2]
```diff
@@ -129,7 +129,55 @@ The design for the demo consists of a number of different entities which play
...
+<!--
+This is an embedded PlantUML config for the overview diagram.
+If you change it, be sure to update the generated diagram by running `plantuml -tsvg  README.md`!
+-->
+@startuml ov
```
Comment (mhutchinson) [inline binary_transparency/firmware/docs/design/README.md:138]: I think overview is a better name than ov. Can we go back to that?

**Review by mhutchinson — COMMENTED** [review]
This is neat!

File: binary_transparency/firmware/docs/design/README.md (hunk 2 above)
Comment (AlCutter) [reply binary_transparency/firmware/docs/design/README.md]: done

File: binary_transparency/firmware/docs/design/README.md (hunk 1 above)
Comment (AlCutter) [reply binary_transparency/firmware/docs/design/README.md:133]: yeah, it works in the preview tool locally but not on GH - I'm going to check it in as is and see what happens on the "real" doc - can double back and sort it out afterwards if it's broken there too.

Comment (mhutchinson) [reply binary_transparency/firmware/docs/design/README.md:133]: https://gist.github.com/noamtamim/f11982b28602bd7e604c233fbe9d910f#gistcomment-3277778

**Review by mhutchinson — APPROVED** [review]
From a quick search I'm pretty convinced you're going to see the UML in the rendered page. But if you want to go for it and sort it out, I won't stop you :-)

**Pull request merged**